### DISCUSSION TITLE
Cast to var

### DISCUSSION THREAD

**Discussion by ekolis** [dotnet/csharplang#8803] [Language Ideas]
This would be a shortcut for casting to the type of a variable - for instance

```
int x = 85;
object y = 42;
x = (var)y;
```

the cast would be equivalent to `(int)y` because `x` is of type `int`.

**Comment by alrz** [comment]
This looks like `downcast` and `upcast` (a static cast) from F#, both of which are target-typed. 

For upcasting, it's recommended to use a helper method
```cs
T StaticCast<T, TDerived>(TDerived o) where TDerived : T => o;
```
But now you need to specify all type arguments. While *partial type inference* (https://github.com/dotnet/csharplang/issues/1349) could help with that, you could hide `TDerived` using a static class,
```cs
static class StaticCast<T> {
   static T From<TDerived>(TDerived o) where TDerived : T => o;
}
```
But once we have *type inference from target-type* (https://github.com/dotnet/csharplang/issues/92) you could use the original method with all type arguments inferred:
```cs
x = Cast(y); // Cast<int, object>(y)
```
which will work for upcasting as well.

**Comment by ronnygunawan** [comment]
Why not write it even shorter:
```cs
x = ()y;
```
for target-typed explicit cast.

**Comment by ekolis** [comment]
x = y; ? What is this, VB? ;)

**Comment by ronnygunawan** [comment]
`x = y;` only works for upcast
